tests/meshTriangle.cpp

```diff
@@ -9,7 +9,7 @@ TEST(MeshTriangle, getBoundingBox) {
   auto a = Vector3f(-1, 0, 3);
   auto b = Vector3f(1, -2, 0);
   auto c = Vector3f(0, 2, -3);
-  auto d = Vector3f(0);
+  auto d = Vector3f(0, 0, 0);
 
   auto diffuse = new Diffuse(new Spectrum(0.0, 0.5, 0.5));
```

Comment:
Easier: Just tie the lifetime of `diffuse` to the scope here by allocating it on the stack:

auto diffuse = Diffuse(new Spectrum(0.0, 0.5, 0.5));
auto tri = MeshTriangle(1, &diffuse, a, b, c, d, d, d);

...

// No delete necessary.

Reply:
Thanks for spotting this and giving me this advice. :smiley_cat:  I still tend to use too complicated solutions... :sweat_smile: